Fix issue with disabled togglecontrol double border

jasmussen opened the pull request:
Fixes #12049.

This PR fixes an issue where a togglecontrol, wrapped in a Disabled component, would show a double border. The issue was that it inherited a border from the parent checkbox style (because the toggle is technically a reskinned checkbox).

Before:

<img width="240" alt="screenshot 2018-11-20 at 10 04 48" src="https://user-images.githubusercontent.com/1204802/48762837-3145fb00-ecac-11e8-8d0d-8a25692b0d8f.png">

After:

<img width="261" alt="screenshot 2018-11-20 at 10 04 34" src="https://user-images.githubusercontent.com/1204802/48762840-330fbe80-ecac-11e8-9c23-55edcad48677.png">

Commit:
Fix issue with disabled togglecontrol double border

Fixes #12049.

This PR fixes an issue where a togglecontrol, wrapped in a Disabled component, would show a double border. The issue was that it inherited a border from the parent checkbox style (because the toggle is technically a reskinned checkbox).

sirreal reviewed (approved):
I didn't recompile styles, but I verified that a toggle control wrapped in `Disabled` does not have the double border when manually adding this style via browser dev tools:

```css
.components-toggle-control .components-base-control__field .components-form-toggle__input {
    border: none;
}
```

Looks like this fixes the issue. Thanks!

jasmussen commented:
Thanks for the review! Not sure why the button doesn't turn green with your approval. But in any case, it's milestoned, so we'll get it in.

simison reviewed (approved):
Compiled and confirmed this fixes it for both content area and the sidebar:

<img width="202" alt="image" src="https://user-images.githubusercontent.com/87168/48769394-37e06c80-ecc4-11e8-979e-6c2626344928.png">

simison commented:
>Not sure why the button doesn't turn green with your approval.

We don't have special rights to this repo. 😉  ✨

jasmussen commented:
CC: @ockham, perhaps the real fix is to add a disabled style to the component itself?

ockham commented:
> CC: @ockham, perhaps the real fix is to add a disabled style to the component itself?

Not sure I understand -- adding `disabled` styling was what I planned to do in the first place: #12105

ockham commented on packages/components/src/toggle-control/style.scss:
```diff
@@ -10,4 +10,9 @@
 	}
+
+	// This overrides a border style that is inherited from parent checkbox styles.
+	.components-form-toggle__input {
```
Since this styles `components-form-toggle__input` -- should the styling actually go to `packages/components/src/form-toggle/style.scss` so it'd also apply to `FormToggle` when not wrapped in a `ToggleControl`? Cf https://github.com/Automattic/wp-calypso/issues/28701

jasmussen replied:
Let me take a look as I address the other item from @simison, thanks!

jasmussen commented:
> Not sure I understand -- adding disabled styling was what I planned to do in the first place: #12105

Sorry I guess I just left this there without too much context :D 

What i meant was that instead of wrapping the FormToggle in a Disabled component to disable it, perhaps having an actual disabled prop on the form toggle would be a better approach? Though I suppose both use cases need to work.

jasmussen committed:
Address feedback.

- Add a better disabled state that works for both checked and unchecked states.
- Move code to the right component.
- Fix some additional bleed issues.

jasmussen commented:
Thanks for the review, everyone. Fixed a few issues, including those mentioned:

![screenshot 2018-11-20 at 15 26 58](https://user-images.githubusercontent.com/1204802/48780145-7e8c9180-ecd9-11e8-9747-fca09a3b9c5f.png)

![screenshot 2018-11-20 at 15 28 07](https://user-images.githubusercontent.com/1204802/48780146-7fbdbe80-ecd9-11e8-8634-0a23b598aafc.png)

ockham reviewed (approved):
Thanks @jasmussen, looks great! Any chance this makes it into 4.5? Would be great for people testing the Jetpack beta after today's code freeze so they won't run into https://github.com/Automattic/wp-calypso/issues/28701 :smile:

jasmussen commented:
Thanks for the review. I think that's up to @youknowriad.

youknowriad reviewed (approved):
How can I go against three approvals?

ockham commented:
Thanks @jasmussen and @youknowriad! :heart:
